src/rules/no-extraneous-dependencies.js

```diff
@@ -1,86 +1,89 @@
 import path from 'path'
 import fs from 'fs'
-import { isArray, isEmpty } from 'lodash'
 import readPkgUp from 'read-pkg-up'
 import minimatch from 'minimatch'
 import resolve from 'eslint-module-utils/resolve'
 import importType from '../core/importType'
 import isStaticRequire from '../core/staticRequire'
 import docsUrl from '../docsUrl'
 
+const CWD = process.cwd()
+
 function hasKeys(obj = {}) {
   return Object.keys(obj).length > 0
 }
 
-function extractDepFields(pkg) {
+function extractDepFields({
+  dependencies = {},
+  devDependencies = {},
+  optionalDependencies = {},
+  peerDependencies = {},
+}) {
   return {
-    dependencies: pkg.dependencies || {},
```

Comment:
this isn't quite the same thing; someone might have `"dependencies": ""` or `"dependencies": null`